### PR TITLE
Deflaking a lot of tests

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/aspnet/AspNetCore-Internal/issues/2633
Fixes https://github.com/aspnet/AspNetCore-Internal/issues/2521
Adds extra logging for https://github.com/aspnet/AspNetCore-Internal/issues/2570
Fixes https://github.com/aspnet/AspNetCore-Internal/issues/2568

### CODE REVIEW

File: src/Hosting/Server.IntegrationTesting/src/Deployers/SelfHostDeployer.cs [hunk 1]
```diff
@@ -55,14 +56,23 @@ public override async Task<DeploymentResult> DeployAsync()
...
+                    try
+                    {
+                        (actualUrl, hostExitToken) = await StartSelfHostAsync();
+                    }
+                    catch (Exception ex)
+                    {
+                        // Retry 3 times and throw
+                        if (i == RetryCount - 1)
+                        {
+                            throw ex;
```
Comment (Tratcher) [inline src/Hosting/Server.IntegrationTesting/src/Deployers/SelfHostDeployer.cs]:
```suggestion
                            throw;
```

File: src/Servers/startvs.cmd [hunk 2]
```diff
@@ -1,3 +1,3 @@
 @ECHO OFF
 
-%~dp0..\..\startvs.cmd %~dp0Kestrel.sln
+%~dp0..\..\startvs.cmd %~dp0FunctionalTests.sln
```
Comment (Tratcher) [inline src/Servers/startvs.cmd]: ?

File: src/Servers/test/FunctionalTests/NtlmAuthenticationTest.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@ public NtlmAuthenticationTests(ITestOutputHelper output) : base(output)
         }
 
         public static TestMatrix TestVariants
-            => TestMatrix.ForServers(ServerType.IISExpress, ServerType.HttpSys, ServerType.Kestrel)
+            => TestMatrix.ForServers(ServerType.HttpSys)
```
Comment (Tratcher) [inline src/Servers/test/FunctionalTests/NtlmAuthenticationTest.cs]: ? The other servers should be tested.

**Comment by anurse** [comment]
"Deflaking" is my favorite work. As is the phrase "a lot of tests". This PR is good PR. Nice PR.

**Comment by jkotalik** [comment]
> This PR is good PR. Nice PR.

approve it then.

**Comment by anurse** [comment]
I Approve.... OF this PR existing ;). It has many codes.

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/FrebTests.cs [hunk 4]
```diff
@@ -76,7 +78,11 @@ public async Task CheckFailedRequestEvents()
             AssertFrebLogs(result, new FrebLogItem("ANCM_INPROC_ASYNC_COMPLETION_COMPLETION", "2"));
         }
 
+        // I think this test is flaky due to freb file not being created quickly enough.
+        // Adding extra logging, marking as flaky, and repeating should help
         [ConditionalFact]
+        [Flaky("https://github.com/aspnet/AspNetCore-Internal/issues/2570", FlakyOn.Helix.All)]
+        [Repeat(10)]
```
Comment (anurse) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/FrebTests.cs]: ~~Remove before merge, yes?~~ Never mind, I see this is about trying to flush out more flaky. Make sure it doesn't significantly increase test run time.

**Review by analogrelay — COMMENTED** [review]
It looks fine to me, but I don't have a huge amount of context here. No red flags from me but I don't feel confident giving it a green one alone ;).

File: src/Servers/IIS/IIS/test/IIS.Tests/ClientDisconnectTests.cs [hunk 5]
```diff
@@ -46,7 +46,8 @@ public async Task WritesSucceedAfterClientDisconnect()
                     await SendContentLength1Post(connection);
                     await requestStartedCompletionSource.Task.DefaultTimeout();
                 }
-                clientDisconnectedCompletionSource.SetResult(true);
+
+                await requestAborted.Task;
```
Comment (davidfowl) [inline src/Servers/IIS/IIS/test/IIS.Tests/ClientDisconnectTests.cs]: timeouts?

**Comment by jkotalik** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=215406&view=logs

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2489

**Comment by jkotalik** [comment]
Bleh, I think I introduced a hanging test with this. I'll investigate it.

**Comment by anurse** [comment]
> hanging test

Oh that's my least favorite word. This is now bad PR.

**Comment by jkotalik** [comment]
I'll fix the self-host deployer later. Something was causing a test to hang. Otherwise, 🆙 📅

File: src/Servers/IIS/IIS/test/IIS.Tests/ClientDisconnectTests.cs [hunk 6]
```diff
@@ -20,19 +19,20 @@ namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests
...
 
             var data = new byte[1024];
             using (var testServer = await TestServer.Create(
                 async ctx =>
                 {
                     requestStartedCompletionSource.SetResult(true);
-                    await clientDisconnectedCompletionSource.Task;
+                    ctx.RequestAborted.Register(() => requestAborted.SetResult(true));
+
+                    await requestAborted.Task;
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/test/IIS.Tests/ClientDisconnectTests.cs]: `DefaultTimeout()`

**Review by BrennanConroy — APPROVED** [review]

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/FrebTests.cs [hunk 7]
```diff
@@ -130,6 +137,7 @@ private IEnumerable<FrebLogItem> GetFrebLogItems(IISDeploymentResult result)
             var xmlFiles = Directory.GetFiles(folderPath).Where(f => f.EndsWith("xml"));
             var frebEvents = new List<FrebLogItem>();
 
+            result.Logger.LogInformation($"Number of freb files available {xmlFiles.Length}.");
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/FrebTests.cs]: Fix!

**Pull request merged**